### PR TITLE
feat: add cross-chain-bridge plugin

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Closes [#279](https://mask.atlassian.net/browse/MF-279)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                            key={bridge?.name}
+                            onClick={() => window.open(bridge?.link)}>
+                            {bridge?.icon}
+                            <div className={classes.bridgeInfo}>
+                                <div className={classes.bridgeName}>
+                                    {bridge?.name}
+                                    {bridge?.isOfficial && <div className={classes.officialTag}>{t.official()}</div>}
+                                </div>
+                                {bridge?.intro && (
+                                    <Typography className={classes.bridgeIntro}>{bridge?.intro}</Typography>
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx]:
```suggestion
                                    <Typography className={classes.bridgeIntro}>{bridge.intro}</Typography>
```

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/MaskIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+</svg> */
+}
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/MaskIcon.tsx]:
```suggestion
```

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                const [open, setOpen] = useState(false)
+                return (
+                    <>
+                        <ApplicationEntry
+                            title="Cross-chain"
+                            disabled={disabled}
+                            icon={new URL('./assets/cross-chain.png', import.meta.url).toString()}
+                            onClick={() => setOpen(true)}
+                        />
+                        {open && <CrossChainBridgeDialog open={open} onClose={() => setOpen(false)} />}
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx]:
```suggestion
                        <CrossChainBridgeDialog open={open} onClose={() => setOpen(false)} />
```

File: packages/plugins/CrossChainBridge/src/constants.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    RainbowBridgeIcon,
+} from './SNSAdaptor/MaskIcon'
+export const PLUGIN_ID = PluginId.CrossChainBridge
+export const PLUGIN_DESCRIPTION = 'An example plugin of Mask Network.'
+export const PLUGIN_NAME = 'Example'
+
+export const CROSS_CHAIN_BRIDGE = [
+    {
+        name: 'CBridge',
+        intro: 'Powered by Celer Network. Support $Mask!',
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/constants.tsx]: i18n

File: packages/plugins/CrossChainBridge/src/locales/en-US.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Example Plugin",
+    "__entry__": "🤔 Example",
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/locales/en-US.json]:
```suggestion
```

File: packages/plugins/CrossChainBridge/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Example plugin
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/README.md]: Please update README.

**Comment by guanbinrui** [comment]
Rename the plugin to `MaskBridge`.

File: packages/plugins/CrossChainBridge/package.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "@masknet/plugin-cross-chain-bridge",
+    "private": true,
+    "main": "./dist/index.js",
+    "types": "./dist/index.d.ts",
+    "dependencies": {
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/package.json:6]: You also need to add `tsconfig.json`.

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            title={t.dialog_title()}
+            open={open}
+            onBack={onClose}>
+            <DialogContent className={classes.content}>
+                <Stack height="100%" spacing={2}>
+                    {bridges?.map((bridge) => (
+                        <div
+                            className={classes.bridgeItem}
+                            key={bridge?.name}
+                            onClick={() => window.open(bridge?.link)}>
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx]: Prefer to use `openWindow`.

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+            DialogProps={{ classes: { paper: classes.paperRoot } }}
+            title={t.dialog_title()}
+            open={open}
+            onBack={onClose}>
+            <DialogContent className={classes.content}>
+                <Stack height="100%" spacing={2}>
+                    {bridges?.map((bridge) => (
+                        <div
+                            className={classes.bridgeItem}
+                            key={bridge?.name}
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx]: Please set up an ID field for each bridge configuration and use it as `key`.

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+                            onClick={() => window.open(bridge?.link)}>
+                            {bridge?.icon}
+                            <div className={classes.bridgeInfo}>
+                                <div className={classes.bridgeName}>
+                                    {bridge?.name}
+                                    {bridge?.isOfficial && <div className={classes.officialTag}>{t.official()}</div>}
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/CrossChainBridgeDialog.tsx]:
```suggestion
                                    {bridge?.isOfficial && <Typography className={classes.officialTag}>{t.official()}</Typography>}
```

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/MaskIcon.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
+interface IconProps {
+    size: string
+}
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/MaskIcon.tsx]:
```suggestion
```

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+import { base } from '../base'
+import { useState } from 'react'
+import { CrossChainBridgeDialog } from './CrossChainBridgeDialog'
+
+const sns: Plugin.SNSAdaptor.Definition = {
+    ...base,
+    init(signal, context) {},
+    ApplicationEntries: [
+        {
+            RenderEntryComponent({ disabled }) {
```
Comment (guanbinrui) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx:12]: What about the previous MaskBridge entry? Will we remove that?

Comment (Randolph314) [reply packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx:12]: replace MaskBridge as CrossChainBridge

Comment (guanbinrui) [reply packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx:12]: ack

**Review by guanbinrui — APPROVED** [review]
LGTM

**Pull request merged**